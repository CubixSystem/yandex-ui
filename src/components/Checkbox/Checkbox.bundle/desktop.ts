import { FC } from 'react';
import { compose, composeU } from '@bem-react/core';

// base
import { ICheckboxProps as ICheckboxDesktopProps, Checkbox as CheckboxDesktop } from '../Checkbox@desktop';
// _lines
import { withLinesMulti } from '../_lines/Checkbox_lines_multi';
import { withLinesOne } from '../_lines/Checkbox_lines_one';
// _size
import { withSizeM } from '../_size/Checkbox_size_m';
import { withSizeS } from '../_size/Checkbox_size_s';
// _theme
import { withThemeNormal } from '../_theme/Checkbox_theme_normal@desktop';
import { withThemePseudo } from '../_theme/Checkbox_theme_pseudo@desktop';
// _view
import { withViewDefault } from '../_view/Checkbox_view_default@desktop';
// _indeterminate
import { withIndeterminate } from '../_indeterminate/Checkbox_indeterminate';

export * from '../Checkbox@desktop';

export interface ICheckboxProps extends ICheckboxDesktopProps {
    lines?: 'multi' | 'one';
    size?: 'm' | 's';
    theme?: 'normal' | 'pseudo';
    view?: 'default';
    indeterminate?: boolean;
}

export const Checkbox = compose(
    withViewDefault,
    withIndeterminate,
    composeU(withLinesMulti, withLinesOne),
    composeU(withSizeM, withSizeS),
    composeU(withThemeNormal, withThemePseudo),
)(CheckboxDesktop) as FC<ICheckboxProps>;
