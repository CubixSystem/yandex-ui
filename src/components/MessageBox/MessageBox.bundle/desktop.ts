import { compose, composeU } from '@bem-react/core';

import { MessageBox as MessageBoxCommon, MessageBoxPopup as MessageBoxPopupCommon } from '../MessageBox@desktop';
import { withViewDefault } from '../_view/MessageBox_view_default';
import { withViewPromo } from '../_view/MessageBox_view_promo';
import { withViewInverse } from '../_view/MessageBox_view_inverse';
import { withSizeL } from '../_size/MessageBox_size_l';
import { withSizeM } from '../_size/MessageBox_size_m';
import { withSizeS } from '../_size/MessageBox_size_s';

export * from '../MessageBox@desktop';

const enhance = compose(
    composeU(withViewDefault, withViewPromo, withViewInverse),
    composeU(withSizeL, withSizeM, withSizeS),
);

export const MessageBox = enhance(MessageBoxCommon);
export const MessageBoxPopup = enhance(MessageBoxPopupCommon);
