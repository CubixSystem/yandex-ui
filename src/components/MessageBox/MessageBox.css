.MessageBox {
    position: relative;

    display: inline-block;

    font-family: var(--control-font-family);
    font-size: var(--message-box-fontSize);
    line-height: var(--message-box-lineHeight);
    user-select: none;

    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

.MessageBox-Content {
    position: relative;

    box-sizing: border-box;
    padding: var(--message-box-content-spaceAll);
}

.MessageBox_layout_plain .MessageBox-Content {
    text-align: center;
}

.MessageBox-Backdrop {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
}

.MessageBox-Backdrop::before {
    display: block;

    width: 100%;
    height: 100%;

    content: '';

    border-radius: var(--message-box-borderRadius);
    box-shadow: 0 6px 20px -4px rgba(0, 0, 0, .7); /* USE_VARS */
}

.MessageBox-Background {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;

    overflow: hidden;

    border-radius: var(--message-box-borderRadius);
}
