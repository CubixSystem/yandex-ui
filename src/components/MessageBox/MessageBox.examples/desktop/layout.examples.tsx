import React from 'react';

import { Button } from '../../../Button/Button.bundle/desktop';
import { MessageBox, Corner } from '../../MessageBox.bundle/desktop';
import { EXAMPLE_DESKTOP_TOKEN, createDecorators, parameters } from '../examples-config';
import { cnTheme } from '../../../../Theme';
import { theme } from '../../../../Theme/presets/default';

export default {
    title: EXAMPLE_DESKTOP_TOKEN,
    decorators: createDecorators({ scope: 'desktop' }),
    parameters,
};

export const Layout = () => {
    return (
        <div className={cnTheme(theme)}>
            <div style={{ padding: '16px' }}>
                <MessageBox
                    view="default"
                    size="m"
                    layout="plain"
                    corner={[
                        <Corner key="corner" width={42} side="top-left">
                            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 49 49">
                                <path
                                    fill="#ffd426"
                                    stroke="#f6cf2e"
                                    d="M17.644 34.455l9.906 13.912 4.949-16.171
                                    16.296-4.911-14.02-9.83.167-16.948-13.614
                                    10.096L5.135 5.039l5.606 16.07L.567
                                    34.62z"
                                />
                            </svg>
                        </Corner>,
                    ]}
                >
                    Жмите сюда, скорей
                </MessageBox>
            </div>
            <div style={{ padding: '16px' }}>
                <MessageBox view="default" size="m" layout="tooltip">
                    Пометить письмо как важное <i key="i">Shift + l</i>
                </MessageBox>
            </div>
            <div style={{ padding: '16px' }}>
                <MessageBox
                    view="default"
                    size="m"
                    layout="functional"
                    onClose={() => {}}
                    actions={
                        <>
                            <Button size="m" view="clear">
                                Нет
                            </Button>
                            <Button size="m" view="action">
                                Да
                            </Button>
                        </>
                    }
                >
                    Согласны?
                </MessageBox>
            </div>
        </div>
    );
};

Layout.story = {
    name: 'layout',
};
