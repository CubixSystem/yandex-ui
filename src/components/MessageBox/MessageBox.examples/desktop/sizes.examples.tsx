import React from 'react';

import { Button } from '../../../Button/Button.bundle/desktop';
import { MessageBox } from '../../MessageBox.bundle/desktop';
import { EXAMPLE_DESKTOP_TOKEN, createDecorators, parameters } from '../examples-config';
import { cnTheme } from '../../../../Theme';
import { theme } from '../../../../Theme/presets/default';

export default {
    title: EXAMPLE_DESKTOP_TOKEN,
    decorators: createDecorators({ scope: 'desktop' }),
    parameters,
};

export const Size = () => {
    return (
        <div className={cnTheme(theme)}>
            <div style={{ padding: '16px' }}>
                <MessageBox
                    onClose={() => null}
                    view="default"
                    size="s"
                    actions={
                        <>
                            <Button size="s" view="action">
                                Понятно
                            </Button>
                        </>
                    }
                >
                    Новый раздел с вашими покупками
                </MessageBox>
            </div>
            <div style={{ padding: '16px' }}>
                <MessageBox
                    onClose={() => null}
                    view="default"
                    size="m"
                    actions={
                        <>
                            <Button size="m" view="action">
                                Понятно
                            </Button>
                        </>
                    }
                >
                    Новая почта <b>squorax@gmail.com</b> привязана к вашему аккаунту
                </MessageBox>
            </div>
            <div style={{ padding: '16px' }}>
                <MessageBox
                    onClose={() => null}
                    view="default"
                    size="l"
                    actions={
                        <>
                            <Button size="l" view="action">
                                Подробнее
                            </Button>
                        </>
                    }
                >
                    Письмо содержит неверную или поддельную информацию об отправителе.
                    Также кто-то мог изменить текст письма после его отправки.
                </MessageBox>
            </div>
        </div>
    );
};

Size.story = {
    name: 'size',
};
