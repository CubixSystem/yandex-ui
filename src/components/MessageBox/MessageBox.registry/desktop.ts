import { Registry } from '@bem-react/di';
import { compose } from '@bem-react/core';

import { Popup as PopupDesktop } from '../../Popup/Popup@desktop';
import { withViewDefault } from '../../Popup/_view/Popup_view_default';
import { withTargetAnchor } from '../../Popup/_target/Popup_target_anchor';
import { withNonvisual } from '../../Popup/_nonvisual/Popup_nonvisual';
import { cnMessageBox } from '../MessageBox';
import { withOutsideClick } from '../../../hocs/withOutsideClick/withOutsideClick';
import { withZIndex } from '../../../hocs/withZIndex/withZIndex';

const Popup = compose(
    withViewDefault,
    withTargetAnchor,
    withNonvisual,
    withOutsideClick,
    withZIndex,
)(PopupDesktop);

export const registry = new Registry({ id: cnMessageBox() });

registry.set('Popup', Popup);
