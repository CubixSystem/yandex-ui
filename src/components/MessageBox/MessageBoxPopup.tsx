import React, { FC, RefObject, useRef } from 'react';
import { ExtractProps } from '@bem-react/core';
import { useComponentRegistry } from '@bem-react/di';

import { Direction } from '../Popup/Popup';
import { MessageBox, MessageBoxProps, cnMessageBox } from './MessageBox';
import { MessageBoxRegistry } from './MessageBox.registry/interface';

type PopupProps = Pick<ExtractProps<MessageBoxRegistry['Popup']>,
    'visible' | 'anchor' | 'scope' | 'onOutsideClick' | 'mainOffset' |
    'secondaryOffset' | 'onOutsideClick' | 'motionless' | 'tailOffset' | 'viewportOffset' | 'zIndexGroupLevel'>;

export type MessageBoxPopupProps = MessageBoxProps &
    PopupProps & {
        /**
         * Направление для раскрытия компонента
         */
        direction?: Direction | Direction[];
    };

export const MessageBoxPopup: FC<MessageBoxPopupProps> = (props) => {
    const {
        anchor,
        direction,
        mainOffset,
        motionless,
        onOutsideClick,
        scope,
        secondaryOffset,
        tailOffset,
        viewportOffset,
        visible,
        zIndexGroupLevel,
        ...restProps
    } = props;
    const { Popup } = useComponentRegistry<MessageBoxRegistry>(cnMessageBox());

    const directions = Array.isArray(direction) ? direction : [direction as Direction];
    const popupInnerRef = useRef<HTMLDivElement>(null);
    const ignoreRefs = [anchor, popupInnerRef].filter(Boolean) as RefObject<HTMLElement>[];

    return (
        <Popup
            anchor={anchor}
            directions={directions}
            innerRef={popupInnerRef}
            ignoreRefs={ignoreRefs}
            mainOffset={mainOffset}
            motionless={motionless}
            nonvisual
            onOutsideClick={onOutsideClick}
            scope={scope}
            secondaryOffset={secondaryOffset}
            tailOffset={tailOffset}
            target="anchor"
            view="default"
            viewportOffset={viewportOffset}
            visible={visible}
            zIndexGroupLevel={zIndexGroupLevel}
        >
            {({ tailRef }) => <MessageBox {...restProps} tailRef={tailRef} />}
        </Popup>
    );
};
