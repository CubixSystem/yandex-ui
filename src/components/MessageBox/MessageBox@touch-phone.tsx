import { withRegistry } from '@bem-react/di';

import { registry } from './MessageBox.registry/touch-phone';
import { MessageBoxPopup as MessageBoxPopupCommon } from './MessageBoxPopup';

export * from './MessageBox';
export { MessageBoxPopupProps } from './MessageBoxPopup';

export const MessageBoxPopup = withRegistry(registry)(MessageBoxPopupCommon);
