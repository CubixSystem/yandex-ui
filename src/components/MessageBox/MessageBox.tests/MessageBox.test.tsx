import React, { createRef } from 'react';
import { mount, render } from 'enzyme';

import { MessageBox as MessageBoxBase } from '../MessageBox';
import { MessageBox, Corner } from '../MessageBox.bundle/desktop';
import { serverEnvironmentSetup, delay } from '../../../internal/utils';

describe('MessageBox', () => {
    describe('server environment', () => {
        serverEnvironmentSetup();

        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<MessageBox className="mix" />)).toMatchSnapshot();
        });
    });

    describe('client environment', () => {
        test('должен вернуть полный шаблон компонента (snapshot)', () => {
            expect(render(<MessageBox className="mix" />)).toMatchSnapshot();
        });

        test('должен быть установлен корректный displayName', () => {
            expect(MessageBoxBase.displayName).toBe('MessageBox');
        });

        test('должен устанавливать ссылку на корневой DOM элемент', async () => {
            const innerRef = createRef<HTMLDivElement>();
            mount(<MessageBox innerRef={innerRef} />);
            await delay(100);
            expect(innerRef.current).not.toBe(null);
        });

        test('должен устанавливать/удалять модификаторы при обновлении через props', () => {
            const wrapper = mount(<MessageBox />);
            expect(wrapper.find('.MessageBox').prop('className')).not.toMatch('_opaque');
            wrapper.setProps({ opaque: true });
            expect(wrapper.find('.MessageBox').prop('className')).toMatch('_opaque');

            wrapper.setProps({ layout: 'tooltip' });
            expect(wrapper.find('.MessageBox').prop('className')).toMatch('_layout_tooltip');
        });

        test('должен рендерить Close элемент (snapshot)', () => {
            expect(render(<MessageBox onClose={() => {}} />)).toMatchSnapshot();
        });
        test('должен рендерить Background элемент (snapshot)', () => {
            expect(render(<MessageBox background={<div children="back" />} />)).toMatchSnapshot();
        });
        test('должен рендерить Corner элемент (snapshot)', () => {
            expect(render(
                <MessageBox
                    corner={
                        <Corner
                            width={42}
                            side="bottom-left"
                            children={<img />}
                        />}
                />,
            )).toMatchSnapshot();
        });
        test('должен рендерить actions элемент (snapshot)', () => {
            expect(render(<MessageBox actions={ <button />} />)).toMatchSnapshot();
        });
    });
});
