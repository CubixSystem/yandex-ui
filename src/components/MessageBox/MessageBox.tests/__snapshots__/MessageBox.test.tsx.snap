// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox client environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain mix"
>
  <div
    class="MessageBox-Backdrop"
  />
  <div
    class="MessageBox-Content"
  />
</div>
`;

exports[`MessageBox client environment должен рендерить Background элемент (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain"
>
  <div
    class="MessageBox-Backdrop"
  >
    <div
      class="MessageBox-Background"
    >
      <div>
        back
      </div>
    </div>
  </div>
  <div
    class="MessageBox-Content"
  />
</div>
`;

exports[`MessageBox client environment должен рендерить Close элемент (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain MessageBox_hasClose"
>
  <div
    class="MessageBox-Backdrop"
  />
  <div
    class="MessageBox-Content"
  >
    <button
      aria-label="Закрыть"
      class="MessageBox-Close"
      type="button"
    >
      <svg
        viewBox="0 0 16 16"
      >
        <path
          d="M8 0a8 8 0 0 0-8 8 8 8 0 0 0 8 8 8 8 0 0 0 8-8 8 8 0 0 0-8-8zM4.995 3.934L8 6.94l3.005-3.005 1.06 1.061L9.06 8l3.006 3.005-1.061 1.06L8 9.062l-3.005 3.005-1.06-1.061L6.938 8 3.934 4.995l1.06-1.06z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`MessageBox client environment должен рендерить Corner элемент (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain"
>
  <div
    class="MessageBox-Backdrop"
  />
  <div
    class="MessageBox-Content"
  />
  <span
    class="MessageBox-Corner"
    style="width:42px;height:42px;left:-14px;bottom:-14px"
  >
    <img />
  </span>
</div>
`;

exports[`MessageBox client environment должен рендерить actions элемент (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain"
>
  <div
    class="MessageBox-Backdrop"
  />
  <div
    class="MessageBox-Content"
  />
  <div
    class="MessageBox-Buttons"
  >
    <button />
  </div>
</div>
`;

exports[`MessageBox server environment должен вернуть полный шаблон компонента (snapshot) 1`] = `
<div
  class="MessageBox MessageBox_layout_plain mix"
>
  <div
    class="MessageBox-Backdrop"
  />
  <div
    class="MessageBox-Content"
  />
</div>
`;
